### DISCUSSION TITLE
question

### DISCUSSION THREAD

**Discussion by JacProsser** [moonstar-x/discord-music-24-7#44] [Q&A] [answered]
just wondering if you could point me to where the code for "pause_on_empty" is located? I'm currently trying to do this in my bot and it isn't working well

**Comment by moonstar-x** [comment, marked as answer]
Hey, I'm sorry but I believe I haven't mentioned that the option is not really implemented yet. I don't know why I thought it was a good idea to put the options before really implementing them.

In any case, the code is in `app.js`, if you import `pauseOnEmpty` from `settings.js` and put

```js
if (!pauseOnEmpty) return;
```

inside the first lines of code for the `voicrStateUpdate` handler, you shouls be able to achieve this.

Additionally, I'm gonna rework on this soon if you'd prefer to wait.

**Reply by JacProsser** [reply]
I have implemented into my own discord boot using my head now so don't worry. thank you anyway!

**Reply by JacProsser** [reply]
also you should look into erela.js or something as it is a good package

**Reply by moonstar-x** [reply]
Alright no problem, and I will check it out, however I'm a but skeptical about using Lavalink since its true power comes from having multiple nodes where the music can be fetched from which isn't free. Thank you anyways for the recommendation though!

**Reply by JacProsser** [reply]
If  you decide to look into it click [here](https://support.something.host/en/article/lavalink-hosting-okm26z/) to access the details of a free to use lavalink server. This is what i personally use and it is pretty good. And if you was thinking this was for your bot then this is my voiceStateUpdate for my personal bot
```js
if (newState.guild.voice.channel.members.size <= 1 && config["music"].pauseOnEmpty && config["music"]["24/7"]) {
        const player = eulogistdev.manager.players.get(newState.guild.id)
        const channel = eulogistdev.channels.cache.get(player.textChannel);
        let pausedEmpty = new MessageEmbed()
            .setAuthor(`Music has been paused | 24/7 Activated: ${config["music"]["24/7"]} | To resume the music please join the VC.`, config["Embed_Defaults"].IconURL)
            .setColor(`${config["Embed_Defaults"].EmbedColour}`)
        player.pause(true);
        channel.send(pausedEmpty)

    } else if (newState.guild.voice.channel.members.size <= 1 && !config["music"].pauseOnEmpty && !config["music"]["24/7"]) {
        const player = eulogistdev.manager.players.get(newState.guild.id)
        const channel = eulogistdev.channels.cache.get(player.textChannel);
        let pausedEmpty = new MessageEmbed()
            .setAuthor(`Music has been paused. 24/7 Activated: ${config["music"]["24/7"]} | i will leave in 1 minute.`, config["Embed_Defaults"].IconURL)
            .setColor(`${config["Embed_Defaults"].EmbedColour}`)
        channel.send(pausedEmpty)
        setTimeout(() => {

            player.destroy()
        }, parseInt(config["music"].leaveDelay))
```

**Reply by moonstar-x** [reply]
Thanks, will do! :)
